packages/vega-parser/src/parsers/guides/axis-labels.js

```diff
@@ -40,30 +41,50 @@ export default function(spec, config, userEncode, dataRef, size, band) {
     offset: extendOffset(band.offset, _('labelOffset'))
   };
 
-  if (isXAxis) {
-    align = labelAlign || (flushOn
-      ? flushExpr(scale, flush, '"left"', '"right"', '"center"')
-      : 'center');
-    baseline = labelBaseline || (orient === Top ? 'bottom' : 'top');
-    offset = !labelAlign;
-  } else {
-    align = labelAlign || (orient === Right ? 'left' : 'right');
-    baseline = labelBaseline || (flushOn
-      ? flushExpr(scale, flush, '"top"', '"bottom"', '"middle"')
-      : 'middle');
-    offset = !labelBaseline;
-  }
-
-  offset = offset && flushOn && flushOffset
-    ? flushExpr(scale, flush, '-(' + flushOffset + ')', flushOffset, 0)
-    : null;
+  if (!isSignal(orient)) {
+    if (isXAxis) {
+      align = labelAlign || (flushOn
+        ? flushExpr(scale, flush, '"left"', '"right"', '"center"')
+        : 'center');
+      baseline = labelBaseline || (orient === Top ? 'bottom' : 'top');
+      offset = !labelAlign;
+    } else {
+      align = labelAlign || (orient === Right ? 'left' : 'right');
+      baseline = labelBaseline || (flushOn
+        ? flushExpr(scale, flush, '"top"', '"bottom"', '"middle"')
+        : 'middle');
+      offset = !labelBaseline;
+    }
+  
+    offset = offset && flushOn && flushOffset
+      ? flushExpr(scale, flush, '-(' + flushOffset + ')', flushOffset, 0)
+      : null;
 
-  encode = {
-    enter: enter = {
+    enter = {
       opacity: zero,
       x: isXAxis ? tickPos : tickSize,
       y: isXAxis ? tickSize : tickPos
-    },
+    }
+  } else {
+    align = labelAlign ||
+      { signal: `(${orient.signal}) === "${Top}" || (${orient.signal}) === "${Bottom}" ? (${flushOn ? flushExpr(scale, flush, '"left"', '"right"', '"center"').signal : "'center'"}) : (${orient.signal}) === "${Right}" ? "left" : "right"`};
```

Comment:
This is really hard to read. Please add line breaks:

```
{ 
  signal: `(${orient.signal}) === "${Top}" || (${orient.signal}) === "${Bottom}" ?` +  

    `(${flushOn ? flushExpr(scale, flush, '"left"', '"right"', '"center"').signal : "'center'"}) :` +  
    `(${orient.signal}) === "${Right}" ? "left" : "right"`
};
```

Same for `baseline`, `offsetExpr`

Reply:
Also, one way to avoid executing orient's signal twice (if it's an expression) is to use 
`indexof(['top', 'bottom'], ${orient.signal}) >= 0`

Reply:
Added the line breaks.

I find that expression harder to read. Maybe it is okay given this is not exposed?

Reply:
Changed to indexof, per discussion below.